py-utils/src/utils/msg_bus/message_bus/bus/bus_frame.py

```diff
@@ -0,0 +1,142 @@
+            "kafka-python": kafkaFactory,
+            "confluent-kafka": ConfluentFactory
+        })
+        self.__load_adaptor(mq_bus_name)
+        if self.config is not None:
+            self.__load_topic()
+
+        self.schema = TopicSchema()
+        self.client_list = []
+
+    def __load_adaptor(self, mq_bus_name):
+        #config
+        factory = self.m_factory("kafka-python")
+        self.config, self.adaptor, self.admin = factory.config, factory.adaptor, factory.admin
+
+
+    def __load_topic(self):
+        # {bus:"kafka", topics:[{name:"Alert", replication_factor:3, policy:"Remove_on_ACK"}], }
+        # will get the schema from config file . convert the string in config.schema
+        # to TopicInMessage using factory patter
+        self.schema = TopicSchema()
+        for t in self.config['topics']:
+            topic = Topic(t)
+            print('*'*10, topic.name)
+            self.schema.set_topic(topic.name, topic)
+
+    def register_client(self, cls):
+        self.client_list.push(cls)
+        pass
+
+    def set_producer(self, Producer):
+        # check MQ and it's configuration here
+        return producer_cls(bootstrap_servers='localhost:9092')
+
+    def set_consumer(self, Consumer):
+        # check MQ and it's configuration here
+        return consumer_cls(bootstrap_servers='localhost:9092', auto_offset_reset='earliest',
+                            consumer_timeout_ms=1000)
+
+    def set_admin(self, Admin):
+        # check MQ and it's configuration here
+        return client_cls()
+
+    def create(self, role):
+        self.role = role
+
+        self.bus_callback.precreate_busclient(self.role)
+        create_busclient = self.adaptor.create(self.role)
+        self.bus_callback.postcreate_busclient(self.role)
+
+        return create_busclient
+
+    @log_decorator
+    def send(self, producer, message):
+        topic = self.schema.get_topic(message, producer)
+        self.bus_callback.pre_send(producer, topic, message)
+
+        all_topic_list = self.get_all_topics()
+        if topic in all_topic_list:
+        #if True:
+            self.adaptor.send(producer, topic, bytes(message.payload, 'utf-8'))
+        else:
+            # logging.debug("Topic not exist. Create the topic before sending")
+            raise KeyError("Topic not exist. Create the topic before sending")
+        # Will post send consider exception too
+        self.bus_callback.post_send(producer, topic, message)
+
+
+    def get_topic(self, client, message):
+        return self.schema.get_topic(client, message)
+
+    def receive(self, consumer ):
+        self.bus_callback.pre_receive(consumer)
+        consumer_obj = self.adaptor.receive(consumer)
+
+        if self.notifier is not None:
+            consumer_obj = self.notifier.get_caller(consumer_obj)
+
+        self.bus_callback.post_receive(consumer)
+        return consumer_obj
+
+    def subscribe(self, consumer, topic, notifier, pattern=None, listener=None):
```

Comment:
Subscribe and unsubscribe was not part of the design. Please defer this to later patches. 

We need a working code without subscription. Subscription will / may be added later.